car.js

```diff
@@ -135,10 +135,38 @@ async function writeStream (stream) {
   return new CarDatastore(reader, writer)
 }
 
+async function traverseBlock (block, get, car, concurrency = 1, seen = new Set()) {
+  const cid = await block.cid()
+  await car.put(cid, block.encodeUnsafe())
+  seen.add(cid.toString('base58btc'))
+  if (cid.codec === 'raw') return
+  const reader = block.reader()
+  const missing = link => !seen.has(link.toString('base58btc'))
+  const links = Array.from(reader.links()).filter(missing).map((([, link]) => link))
+  while (links.length) {
+    const chunk = links.splice(0, concurrency)
+    const blocks = chunk.map(get)
+    while (chunk.length) {
+      const link = chunk.shift()
+      const block = blocks.shift()
+      if (missing(link)) {
+        await traverseBlock(await block, get, car, concurrency, seen)
+      }
+    }
+  }
+}
+
+async function completeGraph (root, get, car) {
```

Comment:
thoughts on moving `root` to last so we can `...root` at some point?

Reply:
i ended up having to add an optional parameter for configurable concurrency.